app/actions/handlers.py

```diff
@@ -349,3 +347,43 @@ async def action_process_observations(integration, action_config: ProcessObserva
             continue  # Keep processing as many files as possible
     logger.info(f"-- Observations processed with success for integration '{integration_id}'.")
     return {'observations_processed': observations_processed}
+
+
+async def action_get_file_status(integration, action_config: GetFileStatusConfig):
+    logger.info(f"Executing get_file_status action with integration {integration} and action_config {action_config}...")
+    integration_id = str(integration.id)
+    file_name = action_config.filename
+    file_metadata = await file_storage.get_file_metadata(
+        integration_id=integration_id,
+        blob_name=file_name
+    )
+    file_status = file_metadata.get("status", None)
+    logger.info(f"-- File status for '{file_name}' in integration '{integration_id}': {file_status} --")
+    return {"file_status": file_status}
+
+
+async def action_set_file_status(integration, action_config: SetFileStatusConfig):
+    logger.info(f"Executing set_file_status action with integration {integration} and action_config {action_config}...")
+    integration_id = str(integration.id)
+    file_name = action_config.filename
+    file_status = action_config.status
+    await file_storage.update_file_metadata(
+        integration_id=integration_id,
+        blob_name=file_name,
+        metadata={"status": file_status}
+    )
```

Comment:
Same here. Aside updating the metadata, we need to [move the file to the right group](https://github.com/PADAS/gundi-integration-ats/blob/f528affd655d90c0718d0678cf727f86028650b6/app/actions/handlers.py#L298) to really change the status.